### PR TITLE
Stop using legacy syntax for parameters on triggerRemoteJob

### PR DESCRIPTION
**Pull request by jsoriano**
String-based syntax for parameters of `triggerRemoteJob` doesn't work on latest Jenkins LTS. Migrate to the new map-based syntax.

Issue found in https://github.com/elastic/elastic-package/pull/928, solution evaluated in https://github.com/elastic/elastic-package/pull/929.

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -456,12 +458,13 @@ def uploadUnpublishedToPackageStorage(builtPackagesPath, packageZips) {
...
-              gs_package_build_zip_path=${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/${packageZip}
-              gs_package_signature_path=${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/${packageZip}.sig
-              dry_run=false""",
-              useCrumbCache: true,
-              useJobInfoCache: true)
+            parameters: [
+              gs_package_build_zip_path: "${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/${packageZip}",
+              gs_package_signature_path: "${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/${packageZip}.sig",
+              dry_run=false,
+            ],
```
Comment (mrodm) [inline .ci/packaging.groovy:465]: notation for dry_run field:
```suggestion
            parameters: [
              gs_package_build_zip_path: "${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/${packageZip}",
              gs_package_signature_path: "${env.PACKAGE_STORAGE_INTERNAL_BUCKET_QUEUE_PUBLISHING_PATH}/${packageZip}.sig",
              dry_run: false,
            ],
```

**Review by mrodm — COMMENTED** [review]
The second usage of the triggerRemoteJob needs to be updated

File: .ci/packaging.groovy (hunk 1 above)
Comment (jsoriano) [reply .ci/packaging.groovy]: Good one, thanks! A bit concerning that CI didn't catch it...

**Comment by jsoriano** [comment]
@axw I guess the CI failure is unrelated?

**Comment by lahsivjar** [comment]
@jsoriano The `TestFleetIntegrationMonitoring` was a flaky test that has been fixed, I have rerun the tests now.

**Pull request merged**